packages/opentelemetry-plugin-ioredis/test/ioredis.test.ts

```diff
@@ -0,0 +1,302 @@
+          client.hset('hash', 'testField', 'testValue', cb),
+      },
+      {
+        description: 'get',
+        command: 'get',
+        method: (cb: ioredisTypes.CallbackFunction<string | null>) =>
+          client.get('test', cb),
+      },
+    ];
+
+    before(done => {
+      client = new ioredis(URL);
+      client.on('error', err => {
+        done(err);
+      });
+      client.on('ready', done);
+    });
+
+    beforeEach(async () => {
+      await client.set('test', 'data');
+      memoryExporter.reset();
+    });
+
+    after(done => {
+      client.quit(done);
+    });
+
+    afterEach(async () => {
+      client.del('hash');
+      memoryExporter.reset();
+    });
+
+    describe('Instrumenting query operations', () => {
+      it('should create a child span for hset promise', done => {
+        const attributes = {
+          ...DEFAULT_ATTRIBUTES,
+          [AttributeNames.DB_TYPE]: 'redis',
+          [AttributeNames.DB_STATEMENT]: 'hset hash random random',
+        };
+        const span = tracer.startSpan('test span');
+        tracer.withSpan(span, async () => {
+          try {
+            await client.hset('hash', 'random', 'random');
+            assert.strictEqual(memoryExporter.getFinishedSpans().length, 1);
+            span.end();
+            const endedSpans = memoryExporter.getFinishedSpans();
+            assert.strictEqual(endedSpans.length, 2);
+            assert.strictEqual(endedSpans[0].name, `hset`);
+            assertionUtils.assertSpan(
+              endedSpans[0],
+              SpanKind.CLIENT,
+              attributes,
+              [],
+              okStatus
+            );
+            assertionUtils.assertPropagation(endedSpans[0], span);
+            done();
+          } catch (error) {
+            assert.ifError(error);
+            done();
+          }
+        });
+      });
+
+      it('should create a child span for get promise', done => {
+        const attributes = {
+          ...DEFAULT_ATTRIBUTES,
+          [AttributeNames.DB_TYPE]: 'redis',
+          [AttributeNames.DB_STATEMENT]: 'get test',
+        };
+        const span = tracer.startSpan('test span');
+        tracer.withSpan(span, async () => {
+          try {
+            const value = await client.get('test');
+            assert.strictEqual(value, 'data');
+            assert.strictEqual(memoryExporter.getFinishedSpans().length, 1);
+            span.end();
+            const endedSpans = memoryExporter.getFinishedSpans();
+            assert.strictEqual(endedSpans.length, 2);
+            assert.strictEqual(endedSpans[0].name, `get`);
+            assertionUtils.assertSpan(
+              endedSpans[0],
+              SpanKind.CLIENT,
+              attributes,
+              [],
+              okStatus
+            );
+            assertionUtils.assertPropagation(endedSpans[0], span);
+            done();
+          } catch (error) {
+            assert.ifError(error);
+            done();
```

Comment:
Nit: move `done()` from "try" and "catch" to "finally"

Reply:
done! ...pun intended :)